synapse/app/homeserver.py

```diff
@@ -401,6 +401,7 @@ def profiled(*args, **kargs):
     start_time = clock.time()
 
     stats = {}
+    stats_process = []
```

Comment:
Why are you using an array here?
Wouldn't it possible to just use an assignment?
The loop will cause multiple entries to be overwritten either way

Reply:
This is getting set from inside a local function, so an assignment won't work, e.g.:

```python
stats_proc = None

def get_proc():
    stats_proc = object()  # This creates a new local variable

get_proc()

assert stats_proc is None
```

Though I think you're right that its a bit unclear at the moment, we generally write:

```python
# Contains the psutil.Process once we've fetched it.
stats_proc = [None]

def get_proc():
    stats_proc[0] = object()
```

Reply:
Ohh good to know. I expected python to behave like other languages...
then nvm